tests/governance/test-question-batch-gate.sh

```
@@ -0,0 +1,183 @@
+  || bad "absent prompt_id" "rc=$RC fired?"
+
+printf '' | bash "$HOOK" >/dev/null 2>&1
+[ $? -eq 0 ] && ok "empty stdin -> exit 0" || bad "empty stdin" "nonzero exit"
+
+printf 'not json at all' | bash "$HOOK" >/dev/null 2>&1
+[ $? -eq 0 ] && ok "non-JSON stdin -> exit 0" || bad "non-JSON stdin" "nonzero exit"
+
+# BLOCKING-1: unset HOME must not abort under `set -u`
+O="$(jq -cn '{session_id:"h",prompt_id:"h2",last_assistant_message:"a?\nb?\nc?"}' \
+  | env -u HOME -u QBG_STATE_DIR PATH="$PATH" bash "$HOOK" 2>&1)"; RC=$?
+[ "$RC" -eq 0 ] && ok "unset HOME -> exit 0 (BLOCKING-1)" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)